File: openbis_api/source/java/ch/ethz/sis/openbis/generic/shared/api/v3/dto/search/AbstractFieldSearchCriterion.java
/*
 * Copyright 2014 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.shared.api.v3.dto.search;

import ch.systemsx.cisd.base.annotation.JsonObject;

@JsonObject("AbstractFieldSearchCriterion")
public abstract class AbstractFieldSearchCriterion<T> extends AbstractSearchCriterion
{

    private static final long serialVersionUID = 1L;

    private String fieldName;

    private SearchFieldType fieldType;

    private T fieldValue;

    AbstractFieldSearchCriterion(String fieldName, SearchFieldType fieldType)
    {
        this.fieldName = fieldName;
        this.fieldType = fieldType;
    }

    public String getFieldName()
    {
        return fieldName;
    }

    public SearchFieldType getFieldType()
    {
        return fieldType;
    }

    public T getFieldValue()
    {
        return fieldValue;
    }

    public void setFieldValue(T value)
    {
        this.fieldValue = value;
    }

    @Override
    public String toString()
    {
        String descriptor = "";
        switch (getFieldType())
        {
            case PROPERTY:
                descriptor = "with property '" + getFieldName() + "'";
                break;
            case ATTRIBUTE:
                descriptor = "with attribute '" + getFieldName() + "'";
                break;
            case ANY_PROPERTY:
                descriptor = "any property";
                break;
            case ANY_FIELD:
                descriptor = "any field";
                break;
        }
        return descriptor + " " + (getFieldValue() == null ? "" : getFieldValue().toString());
    }

}

File: js-test/servers/common/core-plugins/tests/1/as/webapps/openbis-v3-api-test/html/models/dto/search/AbstractFieldSearchCriterion.js
define(["support/stjs"], function (stjs, AbstractSearchCriterion) {
    var AbstractFieldSearchCriterion = function(fieldName, fieldType) {
        AbstractSearchCriterion.call(this);
        this.fieldName = fieldName;
        this.fieldType = fieldType;
    };
    stjs.extend(AbstractFieldSearchCriterion, AbstractSearchCriterion, [AbstractSearchCriterion], function(constructor, prototype) {
        prototype['@type'] = 'AbstractFieldSearchCriterion';
        constructor.serialVersionUID = 1;
        prototype.fieldName = null;
        prototype.fieldType = null;
        prototype.fieldValue = null;
        prototype.getFieldName = function() {
            return this.fieldName;
        };
        prototype.getFieldType = function() {
            return this.fieldType;
        };
        prototype.getFieldValue = function() {
            return this.fieldValue;
        };
        prototype.setFieldValue = function(value) {
            this.fieldValue = value;
        };
        prototype.toString = function() {
            var descriptor = "";
            switch (this.getFieldType()) {
                case SearchFieldType.PROPERTY:
                    descriptor = "with property '" + this.getFieldName() + "'";
                    break;
                case SearchFieldType.ATTRIBUTE:
                    descriptor = "with attribute '" + this.getFieldName() + "'";
                    break;
                case SearchFieldType.ANY_PROPERTY:
                    descriptor = "any property";
                    break;
                case SearchFieldType.ANY_FIELD:
                    descriptor = "any field";
                    break;
            }
            return descriptor + " " + (this.getFieldValue() == null ? "" : this.getFieldValue().toString());
        };
    }, {fieldType: {name: "Enum", arguments: ["SearchFieldType"]}});
    return AbstractFieldSearchCriterion;
})